### ISSUE TITLE
Propagate send errors for some of the discv5 functionality.

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-eth#241]
The [send](https://github.com/status-im/nim-eth/blob/master/eth/p2p/discoveryv5/protocol.nim#L158) proc currently just logs errors through a async callback.

For some procs that use this it doesn't matter much, e.g. `sendWhoareyou`:
- Here you wouldn't care much that it didn't manage to send. Not much you can do about it if it is a network issue. You would just be able to delete the handshake from the table immediately (or not add it in the first place). It would be similar for `Nodes` and `Pong` replies. Retrying sounds useless because by the time it works again timeouts are likely to have passed.

For other procs it might matter more:
- `ping`: Important on (re)validation of nodes in the routing table. You want to be able to differentiate between e.g. a send failure because of the own node network connection, or because of a lost packet.
- `findNode`: Perhaps a little less important, but the lookup could stop early instead of trying many `findNode`s that will all fail.

So basically, `send` would have to propagate the error, e.g. through `Future[Result[...]]`.
Other option would be to stop the loops but that sounds more complex.
